liquigraph-core/src/main/java/org/liquigraph/core/io/lock/LockableConnection.java

```diff
@@ -42,28 +46,41 @@
  *
  * A shutdown hook is executed to remove the lock node if and
  * only if the connection has not been properly closed.
+ *
+ * Please note that any {@link Connection} passed to this decorator
+ * will be set to auto-commit: false`. The auto-commit property
+ * is restored on close. The latter is done in order to minimize
+ * side effects when connections are recycled by a connection pool,
+ * for instance.
  */
 public final class LockableConnection implements Connection {
     private final Connection delegate;
     private final LiquigraphLock lock;
+    private final boolean previousAutoCommit;
 
-    private LockableConnection(Connection delegate, LiquigraphLock lock) {
+    @VisibleForTesting
+    LockableConnection(Connection delegate, LiquigraphLock lock) throws SQLException {
         this.delegate = delegate;
+        this.previousAutoCommit = this.delegate.getAutoCommit();
+        this.delegate.setAutoCommit(false);
         this.lock = lock;
     }
 
     public static LockableConnection acquire(Connection delegate, LiquigraphLock lock) {
-        LockableConnection connection = new LockableConnection(delegate, lock);
+        LockableConnection connection = null;
         try {
+            connection = new LockableConnection(delegate, lock);
             lock.acquire(connection);
             return connection;
-        } catch (RuntimeException e) {
+        } catch (SQLException | RuntimeException e) {
```

Comment:
The whole `catch` clause is not covered. Please don't kill me 😄. The coverage also decreased because covered code was actually removed from `ConnectionConfigurationByUri`.

By playing with the mocks, you can cover most of it:
  - mock the `LiquigraphLock` so it throws an exception when `acquire` is called: `connection` won't be null, and it will be closed
  - have one test let the `close` go through, and another throw a `SQLException`
  - and if you want to cover all the logical branches, even if Coveralls doesn't care, you can have the `Connection` mock throw a `SQLException` on the `setAutoCommit(false)` in an extra test case

And actually, you could leave the constructor private and go through the `LockableConnection::acquire` factory every time, with the `LiquigraphLock` mock not doing anything in the existing tests, though it'd be nice to verify it was called.